man/coap_resource.txt.in

```diff
@@ -20,6 +20,8 @@ coap_delete_all_resources,
 coap_resource_set_mode,
 coap_resource_set_userdata,
 coap_resource_get_userdata
+coap_resource_release_userdata_handler
+coap_resource_get_uri_path,
```

Comment:
Adding `coap_resource_get_uri_path` leads to the following error during install:

```
/usr/bin/install: will not overwrite just-created '/tmp/build/share/man/man3/coap_resource_get_uri_path.3' with 'coap_resource_get_uri_path.3'
```

The reason is that `coap_resource_get_uri_path.3` is included twice in A2X_EXTRA_PAGES because it is already present in `coap_resource_txt.in`. This could be fixed in the `Makefile.am` with the following patch:

```
diff --git a/man/Makefile.am b/man/Makefile.am
index 91b428a8..050e4e9f 100644
--- a/man/Makefile.am
+++ b/man/Makefile.am
@@ -67,8 +67,8 @@ man7_MANS = $(MAN7)
 
 # Man pages built by a2x based on the NAMES section of the .txt file.
 # Note - this list includes all the defined entries, but a2x only builds the first 10.
-A2X_EXTRA_PAGES = @DOLLAR_SIGN@(shell for fil in $(TXT3) ; do sed -ne '/^NAME/,/^SYNOPSIS/p;/^SYNOPSIS/q' $${fil} | \
-       sed -ne '/coap_/{ s/ *, */\n/g; p }' | sed -ne 's/^\(coap_[a-zA-Z_0-9]\+\).*$$/\1.3/p' ; done)
+A2X_EXTRA_PAGES = @DOLLAR_SIGN@(sort $(shell for fil in $(TXT3) ; do sed -ne '/^NAME/,/^SYNOPSIS/p;/^SYNOPSIS/q' $${fil} | \
+       sed -ne '/coap_/{ s/ *, */\n/g; p }' | sed -ne 's/^\(coap_[a-zA-Z_0-9]\+\).*$$/\1.3/p' ; done))
 
 # a2x builds alternative .3 files up to a limit of 10 names from the
 # NAME section, so that 'man' works against the alternative different
```

With this approach, a function name can occur in multiple `.in` manuals but will get installed only once.

Reply:
The issue here with your approach is that (in this case) `man coap_resource_get_uri_path` will return the contents of either `coap_observe.3` or `coap_resource.3` depending on the build order etc.  It is good to be told about the duplication so that it can be fixed.

I have removed the duplication of the coap_resource_get_uri_path entries in the different man pages.

Reply:
Yes, I was thinking about that as well. I agree that the existing approach is superior as long as it is okay for function names to occur only once.